src/test/java/uk/ac/cam/cl/dtg/isaac/api/UsersFacadeIT.java

```diff
@@ -0,0 +1,478 @@
+                        .put("role", targetUser.getRole())
+                        .put("schoolId", targetUser.getSchoolId())
+                )
+                .put("passwordCurrent", JSONObject.NULL);
+
+        HttpServletRequest request = createRequestWithSession();
+        replay(request);
+
+        HttpServletResponse response = createNiceMock(HttpServletResponse.class);
+        replay(response);
+
+        // Act
+        Response createResponse = usersFacade.createOrUpdateUserSettings(request, response, payload.toString());
+
+        // Assert
+        Assertions.assertEquals(Response.Status.UNAUTHORIZED.getStatusCode(), createResponse.getStatus());
+    }
+
+    /**
+     * Checks that a request to update an existing user's password with the correct existing password succeeds.
+     *
+     * @throws Exception - not expected.
+     */
+    @Test
+    public void createOrUpdateEndpoint_updateUserPasswordWithCorrectCurrentPassword_succeedsAndUpdatesExpectedPropertiesOnly() throws Exception {
+        // Arrange
+        // log in as student
+        RegisteredUserDTO targetUser = integrationTestUsers.ERIKA_STUDENT;
+
+        LoginResult studentLogin = loginAs(httpSession, ITConstants.ERIKA_STUDENT_EMAIL, ITConstants.ERIKA_STUDENT_PASSWORD);
+        HttpServletRequest request = createRequestWithCookies(new Cookie[]{studentLogin.cookie});
+        replay(request);
+
+        JSONObject payload = new JSONObject()
+                .put("registeredUser", new JSONObject()
+                        .put("email", targetUser.getEmail())
+                        .put("emailVerificationStatus", targetUser.getEmailVerificationStatus())
+                        .put("familyName", targetUser.getFamilyName())
+                        .put("firstLogin", false)
+                        .put("gender", targetUser.getGender())
+                        .put("givenName", targetUser.getGivenName())
+                        .put("id", targetUser.getId())
+                        .put("lastSeen", "1686558490205")
+                        .put("lastUpdated", "1581680627635")
+                        .put("loggedIn", true)
+                        .put("password", "test12345")
+                        .put("registeredContext", Set.of())
+                        .put("registrationDate", "1564660299981")
+                        .put("role", targetUser.getRole())
+                        .put("schoolId", targetUser.getSchoolId())
+                )
+                .put("passwordCurrent", ITConstants.ERIKA_STUDENT_PASSWORD);
+
+
+        HttpServletResponse response = createNiceMock(HttpServletResponse.class);
+        replay(response);
+
+        // Act
+        Response createResponse = usersFacade.createOrUpdateUserSettings(request, response, payload.toString());
+
+        // Assert
+        // check status code was 'OK'
+        Assertions.assertEquals(Response.Status.OK.getStatusCode(), createResponse.getStatus());
+
```

Comment:
I guess this doesn't check that the password was actually updated. Is there an easier way of doing this than adding a second step to this test where we change the password back to ERIKA_STUDENT_PASSWORD?

Reply:
I've just added a check for an updated hash in the DB, which doesn't quite fit with the other tests (which focus only on what the API returns) but is simplest and makes sense for this one IMO.